### ISSUE TITLE
Implement FOR

### ISSUE THREAD

**Issue by rogerl50** [ArnoldSmith86/virtualtabletop#371]
Edit: How about the following: for existing `clickRoutine`s, `thisButton` and `thisID` are defined to refer to `this` as a collection and a widget. For a `loopRoutine` defined within a `FOR`, we should define `loopWidget` and `loopID` (or some other names) to refer to a collection containing the widget referred to by `loopID`; these are successively defined to be each element of the collection in the invoking `FOR`. So there is no need for extraneous properties in a `FOR`; it looks like
```
{"func":"FOR",
 "collection":"<collection>",
 "loopRoutine":[...]
}
```

--------------------------

Now that we have CALLed routines, and IF/THEN routines, FOR seems like a logical next step. One idea is something like
```
{"func":"FOR",
 "collection":"<collection>",
 "variable":"<variable name>",
 "loopRoutine":[...]
}
```
which would call the `loopRoutine` once for each element of the collection, passing the id of the element as a parameter in `<variable name>`

The implementation requires some of the same changes needed by `IF`, so once that is merged into main, I'll open a PR for this one if it seems like a good idea.

**Comment by larkob** [comment]
I don't quite understand `variable` - what would be the "string name of the element"?

Each item in the collection is an object, so the `loopRoutine` should have a collection for the current object, which it can use for GET and other operations.

**Comment by rogerl50** [comment]
I intended the `loopRoutine` to be passed in turn the `id` of each item in the collection.
